mem0/vector_stores/azure_ai_search.py

```diff
@@ -126,39 +161,41 @@ def insert(self, vectors, payloads=None, ids=None):
     def _build_filter_expression(self, filters):
         filter_conditions = []
         for key, value in filters.items():
-            # If the value is a string, add quotes
             if isinstance(value, str):
                 condition = f"{key} eq '{value}'"
             else:
                 condition = f"{key} eq {value}"
             filter_conditions.append(condition)
```

Comment:
Injecting values into a filter predicate needs sanitizing:
1. For the key, make sure it's just letters and numbers (no quotes, no spaces, etc.)
2. For the value, escape single quotes (replace "'" with "''")
3. For the else case, if the value is not a `str` but not a number either, it'll get converted into a string, so you still want to make sure it doesn't have single quotes inside